sdk/schema/executable.go

```diff
@@ -8,6 +8,13 @@ import (
 	"github.com/1Password/shell-plugins/sdk"
 )
 
+type AllowedCredentials int
+
+const (
+	// DynamicNumber the executable annotated with this value can support a number of credentials which can't be statically determined.
+	DynamicNumber AllowedCredentials = iota
+)
+
```

Comment:
I have a preference to taking this in a separate MR, as it is quite a fundamental change to the plugin schema that shouldn't IMO be hidden under the umbrella of adding a plugin requiring it.

Reply:
Good point. I renamed this PR to reflect that it's mostly about adding some new functionality to the SDK, and Terraform is only a mere example.

Reply:
If the only purpose of Terraform in this PR is to act as an example, then we should probably not introduce the actual plugin yet and instead turn it into a "usage snippet" in the PR description.

Reply:
It's also handy to couple these changes together for when pulling these dependencies into `op`. What is the main reason why we'd like to separate these? 

Given that these changes also kind of make sense together and their size is not too big, it would make the process easier for me to keep them tied together in the same commit.

Reply:
That's fine by me, but then we'll also have to review the Terraform plugin as part of this PR. And I'd then call this PR "Add Terraform plugin".

Reply:
Done, renamed it back to its original name